code/game/machinery/teleporter.dm

```diff
@@ -71,8 +71,8 @@
 			if(!calibrated && prob(30 - ((accurate) * 10))) //oh dear a problem
 				log_game("[M] ([key_name(M)]) was turned into a fly person")
 				if(ishuman(M))//don't remove people from the round randomly you jerks
-					var/mob/living/carbon/human/human = M
-					if(human.dna && human.dna.species.id == "human")
+					var/mob/living/carbon/human = M
+					if(human.dna)
```

Comment:
```suggestion
					if(C.dna && !HAS_TRAIT(C, TRAIT_RADIMMUNE))
```
Otherwise it'll affect golems too.

Reply:
it shouldn't affect golems originally? as far as i know the check searches for human subraces and golems are carbon/golem?

Reply:
No, golems are a species.

Reply:
https://github.com/Citadel-Station-13/Citadel-Station-13/blob/afd2ea7db660431c8752b1af775e811118e29d50/code/modules/mob/living/carbon/human/species_types/golems.dm#L1
Proof

Reply:
i changed the code to suggested changes on my system and ran it through DM and it didn't compile